PolyEngine/Engine/Src/TimeSystem.cpp

```diff
@@ -0,0 +1,60 @@
+#include "EnginePCH.hpp"
+
+#include "TimeSystem.hpp"
+#include <chrono>
+
+using std::chrono::steady_clock;
+using std::chrono::duration_cast;
+using std::chrono::duration;
+
+//------------------------------------------------------------------------------
+void Poly::TimeSystem::TimeUpdatePhase(World * world)
+{
+	TimeWorldComponent* timeComponent = &world->GetTimeWorldComponent();
+
+	//Get new delta time
+	std::chrono::steady_clock::time_point frameTime = steady_clock::now();
+	duration<float> deltaTime = duration_cast<duration<float>>(frameTime - timeComponent->LastFrameTime);
+#ifdef DEBUG
+	deltaTime = deltaTime > duration_cast<duration<float>>(timeComponent.MIN_TIME_STEP_MS) ? timeComponent.MIN_TIME_STEP_MS : deltaTime;
+#endif // DEBUG
+	timeComponent->DeltaTime = deltaTime.count();
+	timeComponent->LastFrameTime = frameTime;
+
+	//Update timers
+	timeComponent->SystemTime.Time += deltaTime;
+	if (!timeComponent->IsPaused)
+	{
+		timeComponent->GameplayTime.Time += deltaTime;
+	}
+	for (auto& timer : timeComponent->Timers)
+	{
+		if (timeComponent->IsPaused && timer.second.IsPausable)
+			continue;
+		timer.second.Time += deltaTime;
+	}
+}
+
+//------------------------------------------------------------------------------
+void Poly::TimeSystem::RegisterTimer(World * world, const size_t id, const bool isPausable)
+{
+	TimeWorldComponent* timeComponent = &world->GetTimeWorldComponent();
+	//if timer with that id exists, do nothing (for now)
+	if (timeComponent->Timers.count(id) == 0)
```

Comment:
Doesn't matter much, but it would be probably slightly faster to do `unordered_map::insert(id, timer)` and then check the result (if the key already exists in map, you get a bool of `false` value)